src/samples/EvalClient/Program.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// See https://aka.ms/new-console-template for more information
+using System.ClientModel;
+using Microsoft.Agents.CopilotStudio.Client;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.AI;
+using Azure.AI.OpenAI;
+using EvalClient;
+
+// Setup the Direct To Engine client example. 
+
+HostApplicationBuilder builder = Host.CreateApplicationBuilder(args);
+
+// Get the configuration settings for the DirectToEngine client from the appsettings.json file.
+SampleConnectionSettings settings = new SampleConnectionSettings(builder.Configuration.GetSection("DirectToEngineSettings"));
+
+
+// Create an http client for use by the DirectToEngine Client and add the token handler to the client.
+builder.Services.AddHttpClient("mcs").ConfigurePrimaryHttpMessageHandler(() => new AddTokenHandler(settings));
+
+// add Settings and an instance of the Direct To engine Copilot Client to the Current services.  
+builder.Services
+    .AddSingleton(settings)
+    .AddTransient<CopilotClient>((s) =>
+        {
+            var logger = s.GetRequiredService<ILoggerFactory>().CreateLogger<CopilotClient>();
+            return new CopilotClient(settings, s.GetRequiredService<IHttpClientFactory>(), logger, "mcs");
+        }
+    )
+    .AddTransient<IChatClient>((s) =>
+        {
+            return new AzureOpenAIClient(new Uri(settings.AzureOpenAiEndpoint), new ApiKeyCredential(settings.AzureOpenAiKey))
+                    .AsChatClient(settings.AzureOpenAiDeployment);        
+        }    
+    )
+    .AddHostedService<EvaluationService>();
+IHost host = builder.Build();
+host.Run();
```

Comment:
The app does not close when the evaluation is done. It remains open after all tests are done. It requires ctrl-C (on Windows) to close the run. This might cause issues when this application is run in a pipeline.

This line can better be replaced with
await host.StartAsync();

This will ensure the app end after running. It use in  pipeline the next tasks will be able to continue.

![image](https://github.com/user-attachments/assets/2302b69f-cfca-470f-94cf-15feaede0a4f)

Reply:
Cool, thanks!!! <3